### PR TITLE
added provider aeternity

### PR DESCRIPTION
**Pull request by marc0olo**
it might be possible that you need to install libsodium on your CI machine in order to make the tests working

### CODE REVIEW

File: provider-aeternity/src/main/java/network/arkane/provider/balance/AeternityBalanceGateway.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import network.arkane.provider.PrecisionUtil;
+import network.arkane.provider.balance.domain.Balance;
+import network.arkane.provider.balance.domain.TokenBalance;
+import network.arkane.provider.chain.SecretType;
+import network.arkane.provider.exceptions.ArkaneException;
+
+import java.math.BigInteger;
+import java.util.List;
+
+public class AeternityBalanceGateway implements BalanceGateway {
```
Comment (Qkyrie) [inline provider-aeternity/src/main/java/network/arkane/provider/balance/AeternityBalanceGateway.java]: @Component

File: provider-aeternity/src/main/java/network/arkane/provider/balance/AeternityBalanceGateway.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    private AccountService accountService;
+
+    public AeternityBalanceGateway(final AccountService accountService) {
```
Comment (Qkyrie) [inline provider-aeternity/src/main/java/network/arkane/provider/balance/AeternityBalanceGateway.java]: Where does AccountService get provided?

File: provider-aeternity/src/main/java/network/arkane/provider/bridge/AeternityTransactionGateway.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import network.arkane.provider.sign.domain.TransactionSignature;
+import org.springframework.stereotype.Component;
+
+@Component
+@Slf4j
+public class AeternityTransactionGateway implements TransactionGateway {
+
+    private TransactionService transactionService;
+
+    public AeternityTransactionGateway(TransactionService transactionService) { this.transactionService = transactionService; }
```
Comment (Qkyrie) [inline provider-aeternity/src/main/java/network/arkane/provider/bridge/AeternityTransactionGateway.java]: Where is an instance of TransactionService provided?

File: provider-aeternity/src/main/java/network/arkane/provider/secret/generation/AeternitySecretGenerator.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package network.arkane.provider.secret.generation;
+
+import com.kryptokrauts.aeternity.sdk.domain.secret.impl.RawKeyPair;
+import com.kryptokrauts.aeternity.sdk.service.keypair.KeyPairService;
+import com.kryptokrauts.aeternity.sdk.service.keypair.KeyPairServiceFactory;
+import network.arkane.provider.chain.SecretType;
+
+public class AeternitySecretGenerator implements SecretGenerator<AeternitySecretKey> {
```
Comment (Qkyrie) [inline provider-aeternity/src/main/java/network/arkane/provider/secret/generation/AeternitySecretGenerator.java]: @Component

File: provider-aeternity/src/main/java/network/arkane/provider/sign/AeternitySpendTransactionSigner.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import lombok.extern.slf4j.Slf4j;
+import network.arkane.provider.secret.generation.AeternitySecretKey;
+import network.arkane.provider.sign.domain.Signature;
+import network.arkane.provider.sign.domain.TransactionSignature;
+import org.bouncycastle.crypto.CryptoException;
+
+import static network.arkane.provider.exceptions.ArkaneException.arkaneException;
+
+@Slf4j
+public class AeternitySpendTransactionSigner implements Signer<AeternitySpendTransactionSignable, AeternitySecretKey> {
```
Comment (Qkyrie) [inline provider-aeternity/src/main/java/network/arkane/provider/sign/AeternitySpendTransactionSigner.java]: @Component

File: provider-aeternity/src/main/java/network/arkane/provider/balance/AeternityBalanceGateway.java (hunk 2 above)
Comment (marc0olo) [reply provider-aeternity/src/main/java/network/arkane/provider/balance/AeternityBalanceGateway.java]: I oriented on the ethereum implementation and thought this is done on your backend and provided through the constructor. isn't that the case?

File: provider-aeternity/src/main/java/network/arkane/provider/bridge/AeternityTransactionGateway.java (hunk 3 above)
Comment (marc0olo) [reply provider-aeternity/src/main/java/network/arkane/provider/bridge/AeternityTransactionGateway.java]: I oriented on the ethereum implementation and thought this is done on your backend and provided through the constructor. isn't that the case?

here it is necessary to say that the TransactionService needs to be configured that the ApiClient connects to the desired node and network.

File: provider-aeternity/src/main/java/network/arkane/provider/balance/AeternityBalanceGateway.java (hunk 2 above)
Comment (marc0olo) [reply provider-aeternity/src/main/java/network/arkane/provider/balance/AeternityBalanceGateway.java]: AccountService is now provided through AeternityServiceAutoConfiguration

File: provider-aeternity/src/main/java/network/arkane/provider/bridge/AeternityTransactionGateway.java (hunk 3 above)
Comment (marc0olo) [reply provider-aeternity/src/main/java/network/arkane/provider/bridge/AeternityTransactionGateway.java]: TransactionService is now provided through AeternityServiceAutoConfiguration

**Pull request merged**